### ISSUE TITLE
Allow adding media from list pages

### ISSUE THREAD

**Issue by oderyn** [IgnisDa/ryot#583]
I have a few books that I cannot find in Open Library or Google Books. I'd like to be able to manually add a book with the relevant metadata, including title, author, and book cover.

I also have several books that don't have book covers. It would be great if I could manually add/upload a book cover as well as any other missing metadata -- description, etc.

**Comment by IgnisDa** [comment]
There is a create media button on the dashboard. Does it not solve your needs?

**Comment by oderyn** [comment]
In terms of being able to add books that I can't find in Openlibrary or Google books. But being able to edit/add to the metadata of books once they've been created would also be helpful.

FWIW, you might consider adding this button to the media lists, too. Maybe something like this:

![Add Media, 1](https://github.com/IgnisDa/ryot/assets/142800967/7ab71cfa-a03c-4c9a-b9ed-ca995a06aec1)

**Comment by IgnisDa** [comment]
I won't be adding an option to update media since it would make the server code infinitely more complex that it already is. If you really need to, I suggest creating custom media for them too.

Add a button there does make sense. Thanks!

**Comment by oderyn** [comment]
Thanks! I'll try the custom create for items missing images, etc. Just trying to get a local Minio server running so I can upload images locally.

**Comment by IgnisDa** [comment]
You need the following env variables:
```yml
      - FILE_STORAGE_S3_ACCESS_KEY_ID=RdddcJMQaJQMdyKM
      - FILE_STORAGE_S3_BUCKET_NAME=ryot
      - FILE_STORAGE_S3_SECRET_ACCESS_KEY=qTuUantpWsdfdsf4QMbigKWiPh8
      - FILE_STORAGE_S3_URL=https://minio.exmaple.com
```

**Comment by oderyn** [comment]
Yeah, I had some trouble getting the access settings right on Minio, which seems ridiculous given how straightforward it ended up being. Now I just need to sort out the structure of the "Specifics" for the graphql endpoint so I can test image uploading (and the Minio integration).

**Comment by IgnisDa** [comment]
The specfics for books can be `{}` if you do not want to add number of pages.

**Comment by oderyn** [comment]
That helped! But, as suspected, the images aren't getting uploaded to Minio.

**Comment by IgnisDa** [comment]
Open the network on the dev tools and try uploading an image. Screenshot the response.

**Comment by oderyn** [comment]
I got it working. :) The Minio API needed to be on https. As soon as I did that, it went right up.

**Comment by IgnisDa** [comment]
Released with `v4.1.5`.
